### DISCUSSION TITLE
custom extension's insertContent command fails?

### DISCUSSION THREAD

**Discussion by iseus** [ueberdosis/tiptap#2367] [General Questions] [answered]
After updating tiptap my custom extension stopped working, I get
```
RangeError: Invalid content for node foo: <>
    at Node.check (tiptap:47872)
```
errors. The extension works with @tiptap/core@2.0.0-beta.152 and generates the error after beta.153.

I see that this change is made here:
https://github.com/ueberdosis/tiptap/commit/113133b74d5ef2852e80bc257defe2a1d675ebbe

The extension:
```
export default Node.create({
    name: 'foo',
    group: 'block',
    content: 'block+',
    atom: true,
    addAttributes() {
        return {
            id: {
                default: null,
            },
            type: {
                default: null,
            },
            title: {
                default: null,
            },
        }
    },
    addCommands() {
        return {
            addFoo: options => ({ commands }) => {
                return commands.insertContent({
                    type: this.name,
                    attrs: options,
                })
            },
        }
    },
    parseHTML() {
        return [{
            tag: 'foo',
        }]
    },
    renderHTML({ HTMLAttributes }) {
        return ['foo', mergeAttributes(HTMLAttributes), 0]
    },
    addNodeView() {
        return VueNodeViewRenderer(FooComponent)
    },
})
```
In the browser if I make a break in Node.check(), the node's content length is 0. I use VueNodeViewRenderer, so I think the problem is somewhere there, the content is not ready at that point to validate.

How should I upgrade my extension to be compatible with the new version again?

**Comment by philippkuehn** [comment, marked as answer]
Hey, it’s because `insertContent` is very strict when using JSON content. You have defined `block+` as content, which is why you need at least _one_ child element. Right now, there are three ways of fixing this:

1. Define a child element in `insertContent`

```js
commands.insertContent({
  type: this.name,
  attrs: options,
  content: [
    {
      type: 'paragraph',
    }
  ],
})
```

2. Or use HTML because it is parsed differently.

```js
commands.insertContent('<foo></foo>')
```

3. Or use `block*` instead so it’s allowed to have zero child elements.

```js
export default Node.create({
  // ...
  content: 'block*',
  // ...
})
```

**Reply by iseus** [reply]
Thank you, I really appreciate your fast answer. I chose the 3rd option because my extension works that way, and it's perfect now. Probably I misunderstood the documentation then.
